### PR TITLE
add github workflow for release

### PR DESCRIPTION
**Pull request by akashg3627**

### CODE REVIEW

File: .github/workflows/build-n-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Set docker repo name
+        run: |
+          echo "DOCKER_REPO=truefoundryuser/notebook-controller" >> $GITHUB_ENV
+
+      - name: Set version tags
+        run: |
+          echo "Ref name is ${{ github.ref_name }}"
+          echo "VERSION_TAG=$(echo ${{ github.ref_name }} | awk '{print substr($0, 2, length($0) - 1)}')" >> $GITHUB_ENV
+
+      - name: Docker login
```
Comment (dunefro) [inline .github/workflows/build-n-publish.yaml:32]: you can use this action `docker/login-action@v2`

File: .github/workflows/build-n-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          docker login -u truefoundry -p "${{ secrets.DOCKERHUB_TOKEN }}"
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push container
+        uses: docker/build-push-action@v2
+        with:
```
Comment (dunefro) [inline .github/workflows/build-n-publish.yaml:41]: You can add support for arm platform as well with something like this
```
      - name: Build and Push image to Docker Hub
        uses: docker/build-push-action@v4
        with:
          platforms: linux/amd64,linux/arm64
          context: .
          push: true
```

File: .github/workflows/build-n-publish.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: Build the Notebook controller for Notebook
+
+on:
+  push:
+    tags:
```
Comment (dunefro) [inline .github/workflows/build-n-publish.yaml:5]: Do we want to publish only on tags, not on push to main ?

File: .github/workflows/build-n-publish.yaml (hunk 2 above)
Comment (dunefro) [reply .github/workflows/build-n-publish.yaml:41]: Make the action `v4`

**Review by dunefro — CHANGES_REQUESTED** [review]

File: .github/workflows/build-n-publish.yaml (hunk 3 above)
Comment (akashg3627) [reply .github/workflows/build-n-publish.yaml:5]: yes

**Review by dunefro — APPROVED** [review]
LGTM

**Pull request merged**